### ISSUE TITLE
Missing JSDoc for parameter in a class doc block

### ISSUE THREAD

**Issue by SergioMorchon** [eslint/eslint#10367]
# Environment
* ESLint Version: 4.19.1
* Parser: babel-eslint@8.2.2

# Configuration

*.eslintrc.yml*:
```
parser: babel-eslint
rules:
  valid-jsdoc: [error, {
    prefer: {
      arg: param,
      argument: param,
      return: returns,
    },
    preferType: {
      Boolean: boolean,
      Number: number,
      String: string,
      object: Object,
    },
    requireReturn: false,
    requireReturnType: false,
    requireReturnDescription: false,
    matchDescription: .*,
    requireParamDescription: false
  }]
```

# Sourcecode

```js
/**
 */
class A {
    b = c => {};
}
```

# What I expect

No JSDoc-related errors.

# What happened

`1:1  error  Missing JSDoc for parameter 'c'  valid-jsdoc` for the class-level JSDoc block, requiring a parameter for a inner method.

# Extra info

Changing that binded function to a class method behaves as expected, without JSDoc-related errors:
```js
/**
 */
class A {
    b() {};
}
```

**Comment by platinumazure** [comment]
Hi @SergioMorchon, thanks for the issue.

I don't think this is something we can support right now, since class properties are still a Stage 3 proposal and are not finalized into the language spec. It would be good to check [eslint-plugin-babel](https://github.com/babel/eslint-plugin-babel) to see if they have a version of the valid-jsdoc rule that works on class property functions.

Once class properties hit Stage 4, we will very likely enhance valid-jsdoc to support this use case.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
